src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -737,10 +737,10 @@ def validFiles(self, files):
                 continue
 
             # Create a LumiList from the WMBS info
-            lumis = []
+            runLumis = {}
             for x in f['LumiList']:
-                lumis.append([str(x['RunNumber']), x['LumiSectionNumber']])
-            fileLumiList = LumiList(lumis=lumis)
+                runLumis[str(x['RunNumber'])] = x['LumiSectionNumber']
```

Comment:
I'm not sure this will work either. Typically the agent has run/lumi pairs and here you will overwrite the each run as a new lumi comes in. (In other words, RunNumber is often duplicated)